lvmd/lvservice.go

```diff
@@ -172,6 +172,98 @@ func (s *lvService) RemoveLV(_ context.Context, req *proto.RemoveLVRequest) (*pr
 	return &proto.Empty{}, nil
 }
 
+func (s *lvService) CreateLVSnapshot(_ context.Context, req *proto.CreateLVSnapshotRequest) (*proto.CreateLVSnapshotResponse, error) {
+
+	snapType := req.GetType()
+	switch snapType {
+	case "thin":
+	case "thick":
+		return nil, status.Error(codes.Unimplemented, "thick snapshot is not implemented yet")
+	default:
+		return nil, status.Error(codes.InvalidArgument, "snapshot type is not supported")
+	}
+
+	dc, err := s.mapper.DeviceClass(req.DeviceClass)
```

Comment:
Would you please add a check if this DeviceClass is thin-volume. If it is not thin-volume, should raise an error in the current implementation. And Would you add the same logic when restoring the snapshot?

Reply:
Since both Snapshot and Restore operations call the same `CreateLVSnapshot` method, the added check will verify the thin device classes for both the scenarios :+1: